lib/fs/basicfs_xattr_bsdish.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2022 The Syncthing Authors.
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//go:build freebsd || netbsd
+// +build freebsd netbsd
+
+package fs
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+func listXattr(path string) ([]string, error) {
+	var attrs []string
+
+	// List the two namespaces explicitly and prefix any results with the
+	// namespace name.
+	namespacePrefixes := [...]string{unix.EXTATTR_NAMESPACE_USER: "user.", unix.EXTATTR_NAMESPACE_SYSTEM: "system."}
+	for _, nsid := range [...]int{unix.EXTATTR_NAMESPACE_USER, unix.EXTATTR_NAMESPACE_SYSTEM} {
+		buf := make([]byte, 1024)
+		size, err := unixLlistxattr(path, buf, nsid)
+		if errors.Is(err, unix.ERANGE) {
+			// Buffer is too small. Try again with a zero sized buffer to get
+			// the size, then allocate a buffer of the correct size.
+			size, err = unixLlistxattr(path, nil, nsid)
+			if err != nil {
+				return nil, fmt.Errorf("Listxattr %q: %w", path, err)
+			}
+			if size > len(buf) {
```

Comment:
Wouldn't you want to a abort if <=? What are the chances of not getting the exact same error again if doing the call again with the same `buf`.

Reply:
So there's a detail in how the syscall works here. If called with a non-zero buffer that isn't big enough, it will return the number of bytes written (== the buffer size) and ERANGE. However if we call it with a zero sized buffer (nil, in this case) it returns the number of bytes actually required. In principle I guess we could skip the `if` and always do the `make` because it should always return a size that's bigger than the buffer we already had...